src/core/Global.ttslua

```diff
@@ -295,28 +303,42 @@ function drawChaosToken(params)
     chaosBag.shuffle()
 
     -- add the token to the list, compute new position based on list length
-    tokenOffset[1] = tokenOffset[1] + (0.17 * #chaosTokens)
+    if params.returnedToken then
+      trackChaosToken(params.returnedToken.getName(), matGUID, "subtract")
+      indexOfReturnedToken = getTokenIndex(params.returnedToken)
+      tokenPosition = params.returnedToken.getPosition()
```

Comment:
Should make these variables local (declare above so that they still work as intended)